Feature/sc 19749/add works on sefaria to author pages in mobile

dcschreiber opened the pull request:
This is not going to be merged before modularization launches. 

Adds "Works on Sefaria" section to author topic pages like in the Web version.
Before: Author pages showed a "Sources" tab with text references mentioning the author.

  Changes

api.js
  - Add with_indexes parameter to topic API call to fetch author indexes

LocalizedStrings.js
  - Add localized string: "Works on Sefaria" / "חיבורים וכתבים בספריא"

TopicPage.js
  - Detect author topics via subclass === 'author'
  - Show "Works on Sefaria" tab (instead of "Sources") when author has indexes
  - Add AuthorWork component using CategoryButton for consistent styling with topic pages
  - Hide search bar for works tab
  - Tapping a work navigates in-app using handleOpenURL

dcschreiber committed:
fear(LocalizedStrings): Added worksOnSefaria to the strings

dcschreiber committed:
feat(api): add with_indexes parameter to topic API

  Add support for fetching author indexes from the topic API.
  - Add with_indexes parameter to topic() function (default: true)
  - Include with_indexes in API URL query string

dcschreiber committed:
feat(TopicPage): display Works on Sefaria for author topics

  - Add detection for author topics (subclass === 'author')
  - Show "Works on Sefaria" tab instead of "Sources" for authors with indexes
  - Add AuthorWork component to render work title and description
  - Hide search bar for works tab (works aren't searchable)

dcschreiber committed:
refactor(TopicPage): Code simplification and cleanup

nsantacruz commented on TopicPage.js:
```diff
@@ -327,8 +327,14 @@ const TopicPage = ({ topic, onBack, openNav, openTopic, showToast, openRef, setT
+  const hasWorks = isAuthor && topicData?.indexes?.length;
   const tabs = [];
-  if (!!topicData && !!topicData.textRefs.length) { tabs.push({text: strings.sources, id: 'sources'}); }
+  if (hasWorks) {
```
now that sheets are out of the app, it seems odd to keep using tabs when there's only one tab.

dcschreiber replied:
I agree but think that may be a bit out of scope for this task. 
Mickey actually coincidently mentioned he is OK with the tab style for this change

dcschreiber committed:
feat(TopicPage): use CategoryButton and in-app navigation for AuthorWork

  - Use CategoryButton component for consistent styling with topic pages
  - Navigate to works in-app using handleOpenURL from ReaderAppContext

dcschreiber committed:
Merge branch 'master' into feature/sc-19749/add-works-on-sefaria-to-author-pages-in-mobile
